geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/tube/TubeSelectProcessTest.scala

```diff
@@ -259,8 +262,141 @@ class TubeSelectProcessTest extends Specification {
+      sf.setDefaultGeometry(WKTUtils.read(f"POINT(-178.5 0)"))
+      sf.setAttribute(DefaultDtgField, new DateTime(f"2011-01-01T00:02:03Z", DateTimeZone.UTC).toDate)
+      sf.setAttribute("type", "d")
+      sf.getUserData()(Hints.USE_PROVIDED_FID) = java.lang.Boolean.TRUE
+      featureCollection.add(sf)
+
+      // write the feature to the store
+      val res = fs.addFeatures(featureCollection)
+
+      // tube features
+      val tubeFeatures = fs.getFeatures(CQL.toFilter("type = 'c'"))
+
+      // result set to tube on
+      val features = fs.getFeatures(CQL.toFilter("type <> 'c'"))
+
+      // get back type b from tube
+      val ts = new TubeSelectProcess()
+      val results = ts.execute(tubeFeatures, features, null, 1L, 1L, 0.0, 5, "line")
+
+      val f = results.features()
+      while (f.hasNext) {
+        val sf = f.next
+        sf.getAttribute("type") mustEqual "d"
+      }
+
+      results.size mustEqual 4
+    }
+
+    "properly interpolate times and geometries" in {
+      val sftName = "tubeTestType"
+      val sft = SimpleFeatureTypes.createType(sftName, s"type:String,$geotimeAttributes")
+
+      val ds = createStore
+
+      ds.createSchema(sft)
+      val fs = ds.getFeatureSource(sftName)
+
+      val featureCollection = new DefaultFeatureCollection(sftName, sft)
+      //TODO: Make data that actually emulated a track, and two alternate tracks, one offset from the other by a small amount of time
+      val calc = new GeodeticCalculator()
+      //Configure track generation for 1 hour at 10.2 meters per second @ 80 degrees starting at -141.0, 35.0
+      var curPoint = WKTUtils.read("POINT(-141.0 35.0)")
+      val timePerPoint = 150000l  //2.5 minutes in milliseconds
+      val speed =.0102d // meters/millisecond
+      val startTime = new DateTime(f"2011-01-01T00:00:00Z", DateTimeZone.UTC).toInstant.getMillis
+      val endTime = new DateTime(f"2011-01-01T01:00:00Z", DateTimeZone.UTC).toInstant.getMillis
+      val heading = 80.0d
+      //Validation values
+      var i=0                     //Index for providing an FID to each point
+      var trackCount = 0          //Count number of track points
+      var poolCount = 0           //Count number of valid pool points
+      var invalidPoolCount = 0    //Count number of invalid pool points (those back in time)
+
+
+      (startTime to endTime).by(timePerPoint).foreach { t =>
+        i+=1
+        val sf = AvroSimpleFeatureFactory.buildAvroFeature(sft,List(), i.toString)
+        sf.setDefaultGeometry(curPoint)
+        sf.setAttribute(DefaultDtgField, new DateTime(t, DateTimeZone.UTC).toDate)
+        sf.getUserData()(Hints.USE_PROVIDED_FID) = java.lang.Boolean.TRUE
+        featureCollection.add(sf)
+        if(t % 600000 == 0 ){         //Create a track point every 10 minutes
+          sf.setAttribute("type","track")
+          trackCount+=1
+        } else if (t % 450000 == 0) { //Create invalid pool points every 7 minutes, but put them back in time.
```

Comment:
[picky] Isn't 450,000 milliseconds actually 7.5 minutes?

Reply:
Yes. I'll update the comment.